kernelci/cli/base.py

```diff
@@ -510,8 +511,12 @@ def __init__(self, path, command, cli_args, section=None):
         *section* is a section name to use in the settings file, to provide a
                   way to have default values for each CLI tool
         """
+        self._deprecated_settings = False
```

Comment:
A nicer way to do this I think would be to have a derived class with the `configparser` code and a factory to create objects either of the main Options class which would be using TOML or the .conf one.  That way you wouldn't have to test `if self._deprecated_settings` in multiple places but have overridden methods instead.  And when the time has come to drop support for `configparser`, we can just delete the derived class.

I think it's fine to have the current approach too, the first objective is to have something functional.  Then maybe this could be reworked a bit if the deprecated code is meant to be supported for a long time to make it easier to maintain.

Reply:
I don't think we should put more effort to rework it atm as `configparser` would be removed eventually.
I agree that let's have this functional first and we can rework it later if required.